sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -62,7 +62,7 @@ object ScalaReflection extends ScalaReflection {
   def dataTypeFor[T : TypeTag]: DataType = dataTypeFor(localTypeOf[T])
 
   private def dataTypeFor(tpe: `Type`): DataType = {
```

Comment:
dataTypeFor can be called like this:

    val TypeRef(_, _, Seq(optType)) = t
    val unwrapped = UnwrapOption(dataTypeFor(optType), inputObject)

So we need to dealias it too.